bson/primitive/decimal.go

```diff
@@ -211,6 +212,22 @@ func (d Decimal128) IsZero() bool {
 	return d.h == 0 && d.l == 0
 }
 
+// MarshalJSON returns Decimal128 as a string.
+func (d Decimal128) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// UnmarshalJSON creates a primitive.DateTime from a JSON string.
+func (d *Decimal128) UnmarshalJSON(b []byte) error {
```

Comment:
As discussed offline, this should also handle the extended JSON format as well as the format generated by `json.Marshal` before this work was done.

Reply:
done. Added Extended JSON support, the format generated before this didn't store anything to unmarshal.